format/FlightSql.proto

```diff
@@ -1797,6 +1797,27 @@ message DoPutUpdateResult {
   int64 record_count = 1;
 }
 
+/* An *optional* response returned when `DoPut` is called with `CommandPreparedStatementQuery`.
+ *
+ * *Note on legacy behavior*: previous versions of the protocol did not return any result for
+ * this command, and that behavior should still be supported by clients. See documentation
+ * of individual fields for more details on expected client behavior in this case. 
+ */
+message DoPutPreparedStatementResult {
+  option (experimental) = true;
+
+  // Represents a (potentially updated) opaque handle for the prepared statement on the server.
+  // Because the handle could potentially be updated, any previous handles for this prepared
+  // statement should be considered invalid, and all subsequent requests for this prepared
+  // statement must use this new handle, if specified.
+  // The updated handle allows implementing query parameters with stateless services
+  // as described in https://github.com/apache/arrow/issues/37720.
```

Comment:
We should not point to GH issue discussions as an explanation of why this exists. The spec should ideally be self-contained.